CRM/Contribute/BAO/Contribution/Utils.php

```diff
@@ -118,13 +118,17 @@ public static function processConfirm(
           $contributionParams['payment_instrument_id'] = 1;
         }
       }
+
+      //For free membership/contribution is always a paid membership/contribution with 0 amount
+      $pending = (empty($paymentParams['is_pay_later']) && $form->_amount > 0.0) ? TRUE : FALSE;
+
       $contribution = CRM_Contribute_Form_Contribution_Confirm::processFormContribution(
         $form,
         $paymentParams,
         NULL,
         $contributionParams,
         $financialType,
-        TRUE,
+        $pending,
```

Comment:
I am 95% of the way to removing pending as a function paramter for processFormContribution - with the idea that they are always pending & confirmed via completetransaction if paid